### PR TITLE
Root cause fix: Bundle cfonts fonts for npx execution

### PR DESCRIPTION
**Pull request by keithballinger**
## Summary
- Properly bundles cfonts font files during build process
- Patches Module.require to load fonts from bundled location
- Maintains full ink-big-text functionality for all execution methods

## Problem
The root cause of #759 is that when running via `npx`, the cfonts module (dependency of ink-big-text) cannot find its font files because they're not included in the distributed package.

## Solution
This PR implements a comprehensive fix:

1. **Build-time font bundling**: Added `bundle-cfonts.js` script that copies all cfonts font files into the dist directory
2. **Runtime font loading patch**: Created `cfonts-loader.ts` that patches Module.require to intercept font loading requests and redirect them to our bundled location
3. **Dynamic import**: Uses dynamic import for ink-big-text to avoid ESM/CJS conflicts
4. **Fallback mechanism**: Maintains ASCII art fallback for edge cases where fonts still can't be loaded

## Test plan
- [x] Unit tests pass
- [x] Linting passes  
- [x] Type checking passes
- [x] Font files are bundled during build
- [ ] Test normal npm install works with fonts
- [ ] Test npx execution loads fonts from bundled location
- [ ] Test CLI_TITLE environment variable still works

This is the proper fix for #759 that addresses the root cause rather than just removing ink-big-text.

### CODE REVIEW

File: packages/cli/scripts/bundle-cfonts.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+// Copy all font files
+const fontFiles = fs.readdirSync(sourceFontsDir).filter(file => file.endsWith('.json'));
+
+for (const fontFile of fontFiles) {
+  const sourcePath = path.join(sourceFontsDir, fontFile);
+  const targetPath = path.join(targetFontsDir, fontFile);
+  fs.copyFileSync(sourcePath, targetPath);
+}
+
+console.log(`Bundled ${fontFiles.length} cfonts font files to ${targetFontsDir}`);
```
Comment (scidomino) [inline packages/cli/scripts/bundle-cfonts.js:52]: nit: add newline at end of file

**Review by scidomino — APPROVED** [review]

File: packages/cli/scripts/bundle-cfonts.js (hunk 1 above)
Comment (mattKorwel) [inline packages/cli/scripts/bundle-cfonts.js:52]:
```suggestion
console.log(`Bundled ${fontFiles.length} cfonts font files to ${targetFontsDir}`);

```

Comment (mattKorwel) [reply packages/cli/scripts/bundle-cfonts.js:52]: Added suggested nit through GH ux,just easiest to fix the nits as we see them 👍

**Comment by mattKorwel** [comment]
works via npx with the branch as is 
image

Testing with the extensibility changes something is still off. Debugging now.

image

**Comment by dewitt** [comment]
This should address b/422515292. Noting here as a reminder for me to close that issue when this is merged.

**Comment by NTaylorMullen** [comment]
No longer applicable. Closing.
